### ISSUE TITLE
Trying to implement Rectified Quadradic Unit layer (ReQU) i.e if x > 0 the output is x^2, otherwise 0. Can anyone tell me if there is any problem with this code. After adding this layer to the network the loss is not dropping, it is constant. While using the inbuilt sigmoid in place of ReQU, the error is dropping. Thanks in advance.

### ISSUE THREAD

**Issue by varun-invent** [torch/torch7#275]
require 'nn'

local ReQU,parent = torch.class('nn.ReQU', 'nn.Module')

function ReQU:__init()
   parent.__init(self)
end

function ReQU:updateOutput(input)

  self.output:resizeAs(input):copy(input)

  self.output[torch.le(input,0)] = 0
  self.output:pow(2) 

  return self.output
end

function ReQU:updateGradInput(input, gradOutput)
  
  self.gradInput:resizeAs(gradOutput):copy(gradOutput)
  

  self.gradInput[torch.le(gradOutput,0)] = 0
  self.gradInput:mul(2)

  return self.gradInput
end

**Comment by varun-invent** [comment]
![plot](https://cloud.githubusercontent.com/assets/12339525/8431359/8ca50b52-1f56-11e5-850f-70458277ebdf.png)

**Comment by soumith** [comment]
Hi, for such questions, the google groups is very helpful:

https://groups.google.com/forum/#!forum/torch7

Now, coming to your implementation, writing gradient tests is a good first step to check if your module is correct or not.
Similar to:
https://github.com/torch/nn/blob/master/test.lua#L203

**Comment by varun-invent** [comment]
Thanks :)
